### PR TITLE
[Ubuntu18] switch to gcc 7 

### PR DESCRIPTION
**Pull request by mathildemerle**
This is the main PR allowing users to compile on Ubuntu 18 with the current gcc (7).

I standardize the way we use ${ep}_c_flags and ${ep}_cxx_flags by default in EP_Initialisation.cmake.

DCMTK switch to 3.6.2.

This is needed for Ubuntu 18 users, for 2.4.0.

Linked PR:
 * https://github.com/FBU-IHU-LIRYC/music-utilities/pull/41
 * https://gitlab.inria.fr/MUSIC/MUSICsoftware/merge_requests/26

:m: :)

### CODE REVIEW

File: patches/QtDCM.patch [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+   ${ITKIOPhilipsREC_LIBRARIES}
+diff --git a/Code/QtDcmMoveScu.cpp b/Code/QtDcmMoveScu.cpp
+index d6ea6fe..2ad049d 100644
+--- a/Code/QtDcmMoveScu.cpp
++++ b/Code/QtDcmMoveScu.cpp
+@@ -407,7 +407,7 @@ void QtDcmMoveScu::addOverrideKey ( QString key )
+         qDebug() << QString ( msg2 );
+     }
+ 
+-    DcmElement *elem = newDicomElement ( tag );
```
Comment (mathildemerle) [inline patches/QtDCM.patch:39]: changes in this file to fit DCMTK 3.6.2

File: patches/vtk5.10.1VS2015.patch [hunk 2]
```diff
@@ -1,5 +1,62 @@
...
+-      _gcc_version "${_gcc_version_info}")
+-  endif()
+-  
+   # GCC visibility support, on by default and in testing.
+   check_cxx_compiler_flag(-fvisibility=hidden HAVE_GCC_VISIBILITY)
+   option(VTK_USE_GCC_VISIBILITY "Use GCC visibility support if available." ON)
+   mark_as_advanced(VTK_USE_GCC_VISIBILITY)
+ 
+-  if(${_gcc_version} VERSION_GREATER 4.2.0 AND BUILD_SHARED_LIBS
++  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.2.0 AND BUILD_SHARED_LIBS
```
Comment (mathildemerle) [inline patches/vtk5.10.1VS2015.patch:54]: we can use directly CMAKE_CXX_COMPILER_VERSION, it avoids problem with gcc 6 and above

File: packaging/unix/MUSIC.sh.in [hunk 3]
```diff
@@ -31,7 +31,7 @@ MEDINRIA_BIN="@MEDINRIA_BIN@"
 #   Set the plugins and library paths.
 
 export MEDINRIA_PLUGIN_PATH="@MEDINRIA_PLUGINS_DIRS@"
-export LD_LIBRARY_PATH=${MEDINRIA_DIR}/lib:${MEDINRIA_DIR}/lib/InsightToolkit:${MEDINRIA_DIR}/lib/vtk-5.10:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${MEDINRIA_DIR}/lib:${MEDINRIA_DIR}/DCMTK/install/lib:$LD_LIBRARY_PATH
```
Comment (mathildemerle) [inline packaging/unix/MUSIC.sh.in]: DCMTK libs were not found

Comment (mathildemerle) [reply packaging/unix/MUSIC.sh.in]: And the others were not used, as in the [VTK7 PR](https://github.com/Inria-Asclepios/medinria-superproject/pull/28).

File: cmake/externals/projects_modules/VTK.cmake [hunk 4]
```diff
@@ -56,11 +56,10 @@ endif()
 ## Add specific cmake arguments for configuration step of the project
 ## #############################################################################
 
-# set compilation flags
+# set additional compilation flags
 if (UNIX)
-  set(${ep}_c_flags "${${ep}_c_flags} -w")
-  set(${ep}_cxx_flags "${${ep}_cxx_flags} -w")
-  set(unix_additional_args -DVTK_USE_NVCONTROL:BOOL=ON)
```
Comment (mathildemerle) [inline cmake/externals/projects_modules/VTK.cmake:63]: that was never used anywhere

File: cmake/externals/projects_modules/ITK.cmake [hunk 5]
```diff
@@ -59,12 +59,6 @@ endif()
 ## Add specific cmake arguments for configuration step of the project
 ## #############################################################################
 
-# set compilation flags
-if (UNIX)
-  set(${ep}_c_flags "${${ep}_c_flags} -w")
-  set(${ep}_cxx_flags "${${ep}_cxx_flags} -w -fpermissive")
```
Comment (mathildemerle) [inline cmake/externals/projects_modules/ITK.cmake:65]: ITK now compile without the permissive flag

File: cmake/externals/projects_modules/DCMTK.cmake [hunk 6]
```diff
@@ -87,13 +80,13 @@ set(cmake_args
   -DDCMTK_WITH_DOXYGEN:BOOL=OFF
   -DDCMTK_WITH_ZLIB:BOOL=OFF    
   -DDCMTK_WITH_OPENSSL:BOOL=OFF 
-  -DDCMTK_WITH_PNG:BOOL=OFF     
+  -DDCMTK_WITH_PNG:BOOL=OFF
   -DDCMTK_WITH_TIFF:BOOL=OFF    
   -DDCMTK_WITH_XML:BOOL=OFF     
   -DDCMTK_OVERWRITE_WIN32_COMPILER_FLAGS:BOOL=OFF
+  -DDCMTK_ENABLE_CXX11:BOOL=ON
```
Comment (mathildemerle) [inline cmake/externals/projects_modules/DCMTK.cmake:87]: needed to compile on ubuntu 18.04

**Review by paulineMig — APPROVED** [review]

**Review by mjuhoor — APPROVED** [review]

**Pull request merged**